### PR TITLE
diophantine: note re parametric soln; remove gcd

### LINKED ISSUE #11697: diophantine: Parametrized solutions of ternary quadratics is incomplete
Comparing solutions generated by brute force to those by sympy.

``` python
IPython console for SymPy 1.0.1.dev (Python 3.5.2-64-bit) (ground types: python)

These commands were executed:
>>> from __future__ import division
>>> from sympy import *
>>> x, y, z, t = symbols('x y z t')
>>> k, m, n = symbols('k m n', integer=True)
>>> f, g, h = symbols('f g h', cls=Function)
>>> init_printing()

Documentation can be found at http://docs.sympy.org/dev


In [1]: init_printing(False)
In [2]: from itertools import *

In [3]: solution = diophantine(x**2 + 3*y**2 - z**2)
In [4]: f = lambdify(symbols('p q'), S(str(solution.pop())))

In [5]: bounds = range(-15,15+1)
In [6]: is_solution = lambda x,y,z: x**2 + 3*y**2 == z**2
In [7]: brute_sols = {xyz for xyz in product(bounds, repeat=3)
   ...:                   if is_solution(*xyz)}

In [8]: coprimes = [pq for pq in product(bounds, bounds) if gcd(*pq) == 1]
In [9]: sympy_sols = {xyz for xyz in starmap(f, coprimes)
   ...:                   if all(x in bounds for x in xyz)}

In [10]: len(brute_sols)
Out[10]: 157
In [11]: len(sympy_sols)
Out[11]: 10
```

157 versus 10.  Alright.  To be honest, I wasn't too confident in my interpretation of the documentation of `parametrize_ternary_quadratic`, so I began trying increasingly weaker interpretations.  _Maybe the output order isn't necessarily `(x,y,z)`?_ Or, _since the signs and GCDs among (x,y,z) in the results don't seem to follow any rhyme or reason, maybe it only aims to produce at least one solution from each "branch" of solutions related by signs and gcds?_

``` python
In [12]: # allow the elements of the output tuple to be interpreted as possibly
In [13]: #  belonging to (x,y,z) in a different order
In [14]: sympy_sols3 = {xyz for xyz in starmap(f, product(bounds, repeat=2))
    ...:                    for xyz in permutations(xyz)
    ...:                    if is_solution(*xyz) # but do make sure!
    ...:                    if all(x in bounds for x in xyz)}
In [15]: len(sympy_sols3)
Out[15]: 19

In [16]: # allow any sign of x,y,z and take arbitrary multiples
In [17]: sympy_sols4 = {tuple(m*s*x for (s,x) in zip(signs, xyz))
    ...:                for xyz in sympy_sols3
    ...:                for signs in product([1,-1], repeat=3)
    ...:                for m in bounds
    ...:                if max(map(abs, xyz)) * m in bounds}
In [18]: len(sympy_sols4)
Out[18]: 109
```

_What could still possibly be missing?_

``` python
In [20]: {tuple(map(abs, xyz)) for xyz in brute_sols - sympy_sols4}
Out[20]: set([(1, 1, 2), (3, 3, 6), (5, 5, 10), (7, 7, 14), (11, 5, 14), (13, 3, 14)])
```

and so in addition to **sloppy coverage of solutions which vary by individual signs of (x,y,z) and by gcd(x,y,z)**, it would appear that **it also misses some sort of _fundamental_ solution at (1,1,2),** which may possibly also generate solutions like (11, 5, 14) and (13, 3, 14).

---

**Aside:** I also looked at the [source cited for the method (Nigel P. Smart)](https://books.google.com/books?id=faYE_41prOkC&lpg=PP1&dq=Nigel%20P.%20Smart%2C%20London%20Mathematical%20Society%20Student%20Texts%2041&pg=PP1#v=onepage&q=Nigel%20P.%20Smart,%20London%20Mathematical%20Society%20Student%20Texts%2041&f=false) and tried to work through his proof, but accomplished naught.  On page 54 he multiplies p and q through "by their common denominator so that they take integer coprime values," and I was unable to figure out a justification for the latter point, as I am unable to prove that their numerators are (or can be taken to be) coprime.

He also introduces some mysterious variable _g_ which _sounds_ like it _might_ address my confusion if it were, say, equal to gcd(p.numer, q.numer)---though I'm not quite sure how it managed to get on _that side_ of the equals sign, and the google books preview ends before it ever gets explicitly assigned a value. :/

### PR DESCRIPTION
**Pull request by smichr**
#### References to other Issues or PRs

closes #11697 

#### Brief description of what is fixed or changed

A docstring has been updated to indicate the relationship between the parameterized solution and the actual solution: the parameterized solution does not produce coprime results from coprime input.

I also noticed that symbolic gcd was not being handled so updated the routine that removes the gcd of parameterized solutions.

#### Release Notes

* solvers
    * the gcd of parameterized ternary solutions is now removed

### CODE REVIEW

**Comment by smichr** [comment]
Unless there are suggestions for changes I plan nothing else for this.

**Comment by jksuom** [comment]
Edited release note entry.

File: sympy/solvers/diophantine.py [hunk 1]
```diff
@@ -2170,17 +2175,35 @@ def parametrize_ternary_quadratic(eq):
...
 
+    >>> from sympy import Tuple, ordered
     >>> from sympy.abc import x, y, z
     >>> from sympy.solvers.diophantine import parametrize_ternary_quadratic
-    >>> parametrize_ternary_quadratic(x**2 + y**2 - z**2)
-    (2*p*q, p**2 - q**2, p**2 + q**2)
+    >>> parametrize_ternary_quadratic(4*x**2 + 2*y**2 - 3*z**2)
+    (2*p**2 - 3*q**2, -4*p**2 + 12*p*q - 6*q**2, 4*p**2 - 8*p*q + 6*q**2)
+
+        More than one solution can be represented by a given ``p`` and
```
Comment (jksuom) [inline sympy/solvers/diophantine.py]: Is the extra indentation necessary?

Comment (smichr) [reply sympy/solvers/diophantine.py]: My feeling was that this is more of a parenthetical comment and I wanted it to appear as such. Perhaps putting it under Notes would be better?

Comment (jksuom) [reply sympy/solvers/diophantine.py]: That might be better unless the presence of examples will confuse Sphinx.

**Comment by jksuom** [comment]
This should clarify the nature of the parametrization, pointing out that primitive solutions are not necessarily among those parametrized by integers.

**Pull request merged**